### ISSUE TITLE
Pin images?

### ISSUE THREAD

**Issue by szaimen** [nextcloud-atomic/distro#1]
Hi, thanks for this interesting cool looking new project! 😊

I had a fast look at it and I would probably recommend to pin the version of the images like so for example:
`nextcloud/aio-nextcloud:20241017_085101`
https://github.com/nextcloud-atomic/distro/blob/2a744631545dec054d37becb6ba4bc1928bf51bb/configs/ncatomic/nc_aio/root_overlay/usr/lib/ncatomic/nextcloud-aio/compose.yaml.j2#L106

Also, I guess it would make sense to create some workflows that updates the images then regularly and also updates the compose.yaml regularly from https://github.com/nextcloud/all-in-one/blob/main/manual-install/latest.yml.

WDYT?

**Comment by szaimen** [comment]
Okay, looks like the file was moved to https://github.com/nextcloud-atomic/distro/blob/main/mkosi.images/ncatomic/mkosi.extra/usr/lib/ncatomic/nc-aio/etc/compose.yaml. Still the same question :)

**Comment by theCalcaholic** [comment]
Hey, sorry for getting back to you so late. I totally missed this issue.

I recently switched from Buildroot/SkiffOS to mkosi for building images, so everything in this repo is basically recreated from scratch.

I actually implemented auto updating the compose.yaml during build yesterday, see [here](https://github.com/nextcloud-atomic/distro/blob/main/mkosi.images/ncatomic/mkosi.sync). Only happens if the environment variable `FETCH_UPSTREAM=yes` is set, though, to avoid any suprises :)

Images will definitely be pinned for any releases. Since I'm already [transforming the compose file](https://github.com/nextcloud-atomic/distro/blob/main/mkosi.images/ncatomic/parse_compose.py), image versions (or variables) should be easy to add.
But as long as it's just me testing things, I'm fine with working with `:latest`

**Comment by szaimen** [comment]
Hm, I suppose it would be better to have a workflow that updates the compose file and creates a pull-request so that you can review the changes before merging and building. However I guess this is up to you :)

**Comment by theCalcaholic** [comment]
> Hm, I suppose it would be better to have a workflow that updates the compose file and creates a pull-request so that you can review the changes before merging and building. However I guess this is up to you :)

Good point, I'll probably implement that at a later point. Right now I'm still in a stage, where I'm mostly working from my laptop and don't have invested much in CI/CD. But I'll set that up once I have the basic building blocks (especially the update system) in place

**Comment by szaimen** [comment]
Btw, do you get my Talk messages from our chat on c.nc.c?
